### PR TITLE
implement a fuller calculation

### LINKED ISSUE #7: Implement a fuller range of on-cost commitment calculations
We would like to be able to calculate on-cost commitments for staff members. Given the following input:

* Current grade and point
* Anniversary date
* Contract end date
* Pension scheme
* 1st August negotiated salary increment which is modelled as 2%
* Any additional (salary change, date) pairs which may happen due to, e.g., change of working hours/re-grading of role, etc.
* (Implicitly) that they are currently employed

Provide an API to calculate the total expected commitment from now until the end date of the contract. The repo has [some documentation](https://github.com/uisautomation/pidash-ucamstaffoncosts/blob/master/doc/salary-calculations.ipynb) on how commitments may be calculated given the information above.

Note that the repo currently does not have the grade and salary point table available and that it should correctly handle contribution points and expected promotion dates where grade (but not necessarily point) will change.

### PR DESCRIPTION
**Pull request by rjw57**
This PR implements a fuller calculation of on-costs. It is *highly* recommended that you read the [accompanying documentation](https://43-133829605-gh.circle-artifacts.com/0/doc/index.html) for this PR as it is quite in-depth.

Closes #7.

### CODE REVIEW

**Comment by rjw57** [comment]
The documentation has now been built by CircleCI at https://43-133829605-gh.circle-artifacts.com/0/doc/index.html

File: ucamstaffoncosts/salary/scales.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    #: Grade 9
+    GRADE_9 = enum.auto()
+
+    #: Grade 10
+    GRADE_10 = enum.auto()
+
+    #: Grade 11
+    GRADE_11 = enum.auto()
+
+    #: Grade 11
```
Comment (abrahammartin) [inline ucamstaffoncosts/salary/scales.py]: Typo

File: ucamstaffoncosts/salary/scales.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    GRADE_12_BAND_1 = enum.auto()
+
+    #: Grade 11
```
Comment (abrahammartin) [inline ucamstaffoncosts/salary/scales.py]: Typo

File: ucamstaffoncosts/salary/scales.py [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    GRADE_12_BAND_2 = enum.auto()
+
+    #: Grade 11
```
Comment (abrahammartin) [inline ucamstaffoncosts/salary/scales.py]: Typo

File: ucamstaffoncosts/salary/scales.py [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+    GRADE_12_BAND_3 = enum.auto()
+
+    #: Grade 11
```
Comment (abrahammartin) [inline ucamstaffoncosts/salary/scales.py]: Typo

File: ucamstaffoncosts/costs.py [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+    #: USS
+    USS = enum.auto()
+
+    #: USS with salary exchange
+    USS_EXCHANGE = enum.auto()
+
+    #: NHS
+    NHS = enum.auto()
+
+
```
Comment (abrahammartin) [inline ucamstaffoncosts/costs.py:217]: we need to add MRC Pension (fixed at 15.9%)

**Review by abrahammartin — CHANGES_REQUESTED** [review]

File: ucamstaffoncosts/costs.py (hunk 5 above)
Comment (rjw57) [reply ucamstaffoncosts/costs.py:217]: I'll move this to #6 since I think it makes sense to address it there.

**Review by abrahammartin — APPROVED** [review]

**Pull request merged**